### PR TITLE
Handle emergency commands

### PR DESCRIPTION
**Pull request by kisslorand**
### Requirements

BTT or MKS TFT

### Description

This PR is a bugfix and an enhancement.

__The bugfix:__
A recent Marlin commit of a PR (https://github.com/MarlinFirmware/Marlin/pull/24761) broke the cancellation of the print when printing from onboard media, the TFT doesn't send any command after cancel procedure is started. This PR fixes it.

__The enhancement:__
This PR brings emergency commands handling. Emergency gcodes are recognized, parsed and treated with priority.

### Benefits:

 - faster parsing of emergency gcodes from any source and almost instant cancellation of an ongoing onboard print

### Related Issues

 - onboard print cancellation is broken by a recent commit of a PR (Marlin)
 - emergency commands are treated like regular ones, without any priority or emergency measures taken
- fixes https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/2640

### CODE REVIEW

**Review by digant73 — CHANGES_REQUESTED** [review]

File: TFT/src/User/API/Printing.c [hunk 1]
```diff
@@ -398,7 +398,7 @@ bool printRemoteStart(const char * filename)
   return true;
 }
 
-bool printStart(void)
+bool preparePrint(void)
```
[the remark this answers is not in the document]
Comment (kisslorand) [reply TFT/src/User/API/Printing.c:401]: Do you print from remote connected to TFT?

File: TFT/src/User/API/Printing.c [hunk 2]
```diff
@@ -507,7 +507,7 @@ void printEnd(void)
     shutdownStart();
 }
 
-void printAbort(void)
+void preparePrintAbort(void)
```
[the remark this answers is not in the document]
Comment (kisslorand) [reply TFT/src/User/API/Printing.c:510]: Fixed.

File: TFT/src/User/API/Printing.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (digant73) [reply TFT/src/User/API/Printing.c]: You are confusing actions controlled by TFT/user with events. E.g. function startPrint is used to start a print. I don't understand your logic to complain about something that was properly written, ordered and properly working for which you also received an explanation about naming convention, logic etc... It's you that are trying to change the code interpreting the logic as you think it's correct while you were missing basic concepts. You could limit the changes only to bug fixes, if any, or new useful features.

[the remark this answers is not in the document]
Comment (kisslorand) [reply TFT/src/User/API/Printing.c]: OK, finally I got it. Function "startPrint" is used to start a print and function "printStart" is used to print a start. Yup, properly named... :)

Joking aside... (deleted a bunch of paragraphs) in this case there's no good or bad, it's just different. If you understand that or not it's fine by me either way.

So, do you print from remote connected to TFT?

[the remark this answers is not in the document]
Comment (digant73) [reply TFT/src/User/API/Printing.c]: I agree. This project is a joke for you. That explains your missing interest on any form of cooperation/collaboration.
I agree. There's no good or bad on that code. That explains you are trying to make changes with no valid reasons at all even when the author of that code highlighted that.
I agree. I don't understand why you don't understand that you literally speak on behalf of the author of some code with the author of that code. You misinterpreted the code (e.g. moved the abort message) but you still pretend to apply changes according to your unclear vision. That is also happening with PRs #2591 and #2601.
I agree. I added the WiFi modules on both my printers and improved/fixed/added the remote control features according to feedbacks/FRs provided by users and the author of ESP3D and I would be happy to continue to use them as they are (as light as possible) as it was made possible for other extra features.

[the remark this answers is not in the document]
Comment (kisslorand) [reply TFT/src/User/API/Printing.c]: You really crossed the line (again), the level of insults is beyond any acceptable level.
Luckily I am mature enough to know when a debate turns into a cockfight and I am withdrawing myself from it.

**Pull request merged**